services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,68 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if (thermalMgrQueueHandle == NULL)
+    return ERR_CODE_INVALID_STATE;
 
+  if (event == NULL)
+    return ERR_CODE_INVALID_ARG;
+
+  if( xQueueSend(thermalMgrQueueHandle,
+                ( void * )event,
+                ( TickType_t ) 10) == errQUEUE_FULL)
+  {
+    return ERR_CODE_QUEUE_FULL;
+  }
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  float tempC;
+  thermal_mgr_event_t eventCall;
+
+  error_code_t errTemp = readTempLM75BD(LM75BD_OBC_I2C_ADDR, &tempC);
+  if (errTemp != ERR_CODE_SUCCESS)
+    printConsole("Error reading Temp in OS Handler");
+    //logging for good measure
+
+  if (tempC > LM75BD_DEFAULT_HYST_THRESH)
+    eventCall.type = OVER_TMP_EVENT_CMD;
+  else
+    eventCall.type = SAFE_OPERATION_CMD;
+
+  error_code_t errEvent = thermalMgrSendEvent(&eventCall);
+  if (errEvent != ERR_CODE_SUCCESS)
+    printConsole("Error sending event in OS Handler");
+    //logging for good measure
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  float tempC = 0;
+  thermal_mgr_event_t eventIn;
+  error_code_t errIn;
+
   while (1) {
-    
+    if(xQueueReceive(thermalMgrQueueHandle,
+                     &eventIn,
+                     (TickType_t) 10) == pdTRUE)
```

Comment:
you could make this block time longer since your only expecting to receive an event every couple of seconds